tests/test_sampling.py

```diff
@@ -172,3 +182,12 @@ def test_sampling_with_deterministics_in_nested_models(
     )
     for deterministic, (inputs, op) in deterministic_mapping.items():
         np.testing.assert_allclose(trace[deterministic], op(*[trace[i] for i in inputs]), rtol=1e-6)
+
+
+def test_sampling_with_no_free_rvs(simple_model_no_free_rvs):
+    model = simple_model_no_free_rvs()
+    with pytest.raises(ValueError) as error:
+        trace, stats = pm.inference.sampling.sample(
+            model=model, num_samples=1, num_chains=1, burn_in=1
+        )
+    error.match("The `state` does not contain any `unobserved` values")
```

Comment:
If you're updating the message above, I am ok not validating the error message, since that can be brittle.